lib/seek/publishing/publishing_common.rb

```diff
@@ -78,12 +83,23 @@
       end
 
       def waiting_approval_assets
-        @waiting_approval_assets = ResourcePublishLog.waiting_approval_assets_for(current_user)
+        @requested_approval_assets = ResourcePublishLog.requested_approval_assets_for_user(current_user)
+        @waiting_approval_assets = ResourcePublishLog.waiting_approval_assets(@requested_approval_assets)
+        @rejected_assets = ResourcePublishLog.rejected_assets(@requested_approval_assets)
         respond_to do |format|
           format.html { render template: 'assets/publishing/waiting_approval_assets' }
         end
       end
 
+      def cancel_publishing_request
+        asset = params[:asset_class].classify.constantize.find(params[:asset_id])
```

Comment:
regarding this, I think it is ok for now, but may be an issue if the code afterwards is changed in the future. 
Would the route to cancelling the request always come via the controller that matches the asset class ?

Reply:
`publish_auth` is trigger first, and checks `@asset` which relies on the current controller

Reply:
unless `cancel_publishing_request ` always comes via the people controller?

Reply:
cancel_publishing_request  always comes via the people controller, and I added it to the publish_auth to prevent other people being able to cancel the publish request... But I realised that my test cases were incomplete, and indeed you could cancel someone else's requests...
I tested for:
```
login_as(gatekeeper)
    get :cancel_publishing_request, params: { id: a_person,
                                              asset_id: df.id,
                                              asset_class: df.class }
```
which prevents access because it's getting the `a_person` path, but if it is modified to:
```
login_as(gatekeeper)
    get :cancel_publishing_request, params: { id: gatekeeper,
                                              asset_id: df.id,
                                              asset_class: df.class }
```
The request is cancelled, and it should not be able to... I'm fixing this now...

Reply:
I've set it so that anyone with manage rights can cancel the publish request... see [d54c00f](https://github.com/seek4science/seek/pull/1484/commits/d54c00f191273ad8c8ccb8c9ac8839824f7dc2b9).